client/src/GameState.ts

```diff
@@ -64,31 +70,131 @@ export class GameState {
 
         this.socket.on("playerMove", (playerId, state) => {
             const i = this.getPlayerIndex(playerId);
-            Tetromino.updateFromState(this.otherTetrominoes[i], state, i + 1);
+            this.otherTetrominoes[i].updateFromState(state, i + 1);
         });
 
         this.socket.on("playerPlace", (playerId, state) => {
-            if (playerId == this.playerId) {
-                // the placing event is emitted by this very client, who already handled the local board during emitting
-                return;
-            }
-
             // place the tetro on our board
             const i = this.getPlayerIndex(playerId);
             const tetroToPlace = this.otherTetrominoes[i];
-
-            Tetromino.updateFromState(tetroToPlace, state, i + 1);
+            tetroToPlace.updateFromState(state, i + 1);
             this.placeTetromino(tetroToPlace);
+            this.otherTetrominoes[i] = tetroToPlace;
+
+            // if the other tetromino is bumping into us, freeze ours too.
+            if (
+                this.adjacentWithPlayers(this.currentTetromino, [
+                    this.otherTetrominoes[i],
+                ])
+            ) {
+                this.emitAndPlaceCurrentTetromino();
+            }
```

Comment:
This still leaves us open to one tetromino freezing but not the other... I think we saw that today, perhaps we should do some testing to see how easily that happen.

If we wanted to, I think we could enforce a "both or none" behavior either by `playerPlace` having the state + id of both involved parties or by requiring a pair of playerPlaces to occur for placing to occur... but that also could get messy.